### PR TITLE
✨ Enable OpenSSF Scorecard code scanning alert and badge

### PR DESCRIPTION
**Pull request by joycebrum**
This Pull Request intends to active the OpenSSF Scorecard, a tool to search for security risks through a series of automated checks.

Each automated task returns a score out of ten and a risk level, with remediation prompts to help you fix the problem. Besides, the badge shown in the readme file shows the score of the project.

I think it would be a very useful tool to improve the security of this project.

Any doubts or questions feel free to reach me out.

### CODE REVIEW

File: .github/workflows/scorecards.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    name: Scorecards analysis
+    runs-on: ubuntu-latest
+    permissions:
+      # Needed to upload the results to code-scanning dashboard.
+      security-events: write
+      # Used to receive a badge. (Upcoming feature)
+      id-token: write
+      # Needs for private repositories.
+      contents: read
+      actions: read
```
Comment (woodruffw) [inline .github/workflows/scorecards.yml:24]: Making sure I understand: do we actually need these permissions? The comment makes it sound like they're only needed for private repos, and this one is public.

File: .github/workflows/scorecards.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      - name: "Run analysis"
+        uses: ossf/scorecard-action@f437b3cc043103683806977b725b99217c67bd46 # tag=v2.0.1
+        with:
+          results_file: results.sarif
+          results_format: sarif
+          # (Optional) Read-only PAT token. Uncomment the `repo_token` line below if:
+          # - you want to enable the Branch-Protection check on a *public* repository, or
+          # - you are installing Scorecards on a *private* repository
+          # To create the PAT, follow the steps in https://github.com/ossf/scorecard-action#authentication-with-pat.
+          # repo_token: ${{ secrets.SCORECARD_READ_TOKEN }}
```
Comment (woodruffw) [inline .github/workflows/scorecards.yml:41]: I think these lines can be removed, if they aren't applicable to this particular deployment.

**Comment by woodruffw** [comment]
Thanks for adding this! It looks great overall; I left two small nits about comments/permissions.

File: .github/workflows/scorecards.yml (hunk 1 above)
Comment (joycebrum) [reply .github/workflows/scorecards.yml]: Yeah, I removed them and it still worked fine

**Review by woodruffw — APPROVED** [review]
LGTM, thanks for adding this!

**Pull request merged**